config/initializers/open_id_connect.rb

```diff
@@ -3,3 +3,57 @@
 Rack::OAuth2.logger = Rails.logger
 # Webfinger.logger = Rails.logger
 SWD.logger = Rails.logger
+
+# the openid_connect gem does not support
+# jwt format in the userinfo call.
+# A PR is open to improve the situation
+# https://github.com/nov/openid_connect/pull/54
+module OpenIDConnect
+  class AccessToken < Rack::OAuth2::AccessToken::Bearer
+    private
+
+    def jwk_loader
+      JSON.parse(URI.parse(ENV['AGENT_CONNECT_JWKS']).read).deep_symbolize_keys
+    end
+
+    def decode_jwt(requested_host, jwt)
+      agent_connect_host = URI.parse(ENV['AGENT_CONNECT_BASE_URL']).host
+
+      if requested_host == agent_connect_host
+        JWT.decode(jwt, nil, true, { algorithms: ['ES256'], jwks: jwk_loader })[0]
```

Comment:
```suggestion
        JWT.decode(jwt, key=nil, verify=true, { algorithms: ['ES256'], jwks: jwk_loader })[0]
```
juste savoir qui est quoi